tritium-lib/src/main/java/com/palantir/tritium/io/InstrumentedInputStream.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.io;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.palantir.logsafe.Preconditions;
+import java.io.InputStream;
+
+final class InstrumentedInputStream extends ForwardingInputStream {
+    private final Meter bytes;
+    private final Histogram throughput;
+    private long start;
```

Comment:
concurrent stream access never works great, but we may want to move `start` to a parameter instead rather than object field.

Reply:
Yeah, are you thinking passing `startNanos` as another arg to `after`?

From API consumer perspective, I need to JavaDoc these to make it clear on the args as there will be both `long`

Reply:
If we keep the throughput histograms, I think that would be cleaner.  I commented about some questionable aspects of the throughput histograms given we don't know which operations actually flush and incur cost, or how much data is included within those operations. Perhaps if we do keep the throughput histograms, we should instead accumulate total bytes written for the lifespan of the stream, and sum all the time spent in read/write/flush/close.